kornia/contrib/kmeans.py

```diff
@@ -0,0 +1,175 @@
+# based on https://github.com/subhadarship/kmeans_pytorch
+
+from __future__ import annotations
+
+import torch
+
+from kornia.core import Tensor
+from kornia.core.check import KORNIA_CHECK, KORNIA_CHECK_SHAPE
+
+
+class KMeans:
+    """Implements the kmeans clustering algorithm with euclidean distance as similarity measure.
+
+    Args:
+        num_clusters: number of clusters the data has to be assigned to
+        cluster_centers: tensor of starting cluster centres can be passed instead of num_clusters
+        tolerance: float value. the algorithm terminates if the shift in centers is less than tolerance
+        max_iterations: number of iterations to run the algorithm for
+        device: the device to which all tensors are moved to and computed
+        seed: number to set torch manual seed for reproducibility
+
+    .. code-block:: python
+        kmeans = kornia.contrib.KMeans(3, None, 10e-4, 100, device, 0)
+        kmeans.fit(torch.rand((1000, 5))
+        predictions = kmeans.predict(torch.rand((10, 5)))
+    """
+
+    def __init__(
+        self,
+        num_clusters: int,
+        cluster_centers: Tensor,
+        tolerance: float = 10e-4,
+        max_iterations: int = 0,
+        device: torch.device | None = None,
```

Comment:
i don't think this should be included here -- since it should be propagated with `X`